DXMainClient/DXGUI/Multiplayer/ChatListBox.cs

```diff
@@ -28,7 +32,45 @@ private void ChatListBox_DoubleLeftClick(object sender, EventArgs e)
             if (link == null)
                 return;
 
-            ProcessLauncher.StartShellProcess(link);
+            bool regExpResult = _regexp.Match(link).Success;
+
+            foreach (var elem in ClientConfiguration.Instance.TrustedDomains)
+            {
+                if (string.IsNullOrEmpty(elem))
+                    continue;
+
+                regExpResult = regExpResult || link.Contains(elem, StringComparison.CurrentCultureIgnoreCase);
```

Comment:
You need to extract the domain from the url, not a `string.Contains`.

e.g., `http://malicious.com/cncnet.org/`